module-apps/application-onboarding/ApplicationOnBoarding.cpp

```diff
@@ -93,8 +95,17 @@ namespace app
             return sys::msgHandled();
         });
 
+        connect(typeid(locks::SimSwitched), [&](sys::Message *msg) {
+            auto simState = Store::GSM::get()->sim;
+            if (simState == Store::GSM::SIM::SIM1 || simState == Store::GSM::SIM::SIM2) {
+                phoneLockSubject.setPhoneLock();
+                return sys::msgHandled();
+            }
+            return sys::msgNotHandled();
```

Comment:
shouldn't the sim handling be rewritten to don't let the message with no sim arrived?

Reply:
In this case, it won't change anything. And rewriting the logic will not be so easy because when we do not have the "no SIM" status, pop-ups with errors from the modem will appear at the start if the user does not have any card inserted.

Reply:
alright then